mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/bridge/builtin/impl/DefaultEnumIdentifierBridge.java

```diff
@@ -32,4 +32,9 @@ public T fromDocumentIdentifier(String documentIdentifier, IdentifierBridgeFromD
 		return Enum.valueOf( enumType, documentIdentifier );
 	}
 
+	@Override
+	public boolean isCompatibleWith(IdentifierBridge<?> bridge) {
+		return false;
```

Comment:
See `org.hibernate.search.mapper.pojo.bridge.builtin.impl.DefaultEnumValueBridge#isCompatibleWith` for a more appropriate implementation.